### PR TITLE
fixed algorithm to loop through each pixel of each quadrant

### PR DESCRIPTION
**Pull request by taliayahav**

### CODE REVIEW

File: src/main/java/yahav/photomosaic/ImagePixels.java [hunk 1]
```diff
@@ -19,39 +17,41 @@
...
         PixelReader pixelReader = image.getPixelReader();
-        for (int squareX = 0; squareX < height; squareX += xInc) {
-            for (int squareY = 0; squareY < width; squareY += yInc) {
-                for (int x = 0; x < squareX; x++) {
-                    for (int y = 0; y < squareY; y++) {
+        for (int squareX = 0; squareX < width; squareX += xInc) {
+            for (int squareY = 0; squareY < height; squareY += yInc) { //should loop through all pixels in each quadrant
+                for (int x = 0; x < squareX-50; x++) {
+                    //this is always starting at 0,0
+                    // but need to start at next quadrant
```
Comment (andrewoid) [inline src/main/java/yahav/photomosaic/ImagePixels.java]: You put this comment in here but you didn't bother actually fixing it.

File: src/main/java/yahav/photomosaic/ImagePixels.java [hunk 2]
```diff
@@ -19,39 +17,41 @@
...
+                    for (int y = 0; y < squareY-50; y++) {
```
Comment (andrewoid) [inline src/main/java/yahav/photomosaic/ImagePixels.java]: What is "50" here for?

File: src/main/java/yahav/photomosaic/ImagePixels.java [hunk 3]
```diff
@@ -96,12 +97,13 @@ public double colorDistance(Color firstColor, Color secondColor){
     }
     public void closestColorDiff(){
```
Comment (andrewoid) [inline src/main/java/yahav/photomosaic/ImagePixels.java]: What does this method do?

Comment (taliayahav) [reply src/main/java/yahav/photomosaic/ImagePixels.java]: this method is supposed to find the closest matching average color between the input image and source images

File: src/main/java/yahav/photomosaic/ImagePixels.java [hunk 4]
```diff
@@ -2,106 +2,119 @@
...
         double greenPixels = 0;
         double bluePixels = 0;
-        double numPixels = 0;
+        Color quadrantColorTotal;
         PixelReader pixelReader = image.getPixelReader();
-        for (int squareX = 0; squareX < height; squareX += xInc) {
-            for (int squareY = 0; squareY < width; squareY += yInc) {
-                for (int x = 0; x < squareX; x++) {
-                    for (int y = 0; y < squareY; y++) {
+        int squareX, squareY = 0;
```
Comment (andrewoid) [inline src/main/java/yahav/photomosaic/ImagePixels.java:39]: define these in the for loop.

File: src/main/java/yahav/photomosaic/ImagePixels.java [hunk 5]
```diff
@@ -2,106 +2,119 @@
...
+        FileInputStream input = new FileInputStream("src/main/resources/google.png");
+        Image image = new Image(input);
         List<Color> squareColors = new ArrayList<>();
         int width = (int) image.getWidth();
         int height = (int) image.getHeight();
-        int squares = 50;
-        int xInc = width / squares;
-        int yInc = height / squares;
+        final int SQUARES = 50;
+        final int quadrantSize = SQUARES * SQUARES;
```
Comment (andrewoid) [inline src/main/java/yahav/photomosaic/ImagePixels.java:33]: This isn't right.
Imagine an image that's 100x100.
How many squares will there be? How many pixels in each square?

File: src/main/java/yahav/photomosaic/ImagePixels.java [hunk 6]
```diff
@@ -2,106 +2,119 @@
...
+import java.io.FileNotFoundException;
 import java.util.ArrayList;
 import java.util.List;
 
-public class ImagePixels{
+public class ImagePixels {
 
     public List<Color> squarePixelColors;
-    public List<Color> allImgClrs;
+    public List<Color> allImageColors;
```
Comment (andrewoid) [inline src/main/java/yahav/photomosaic/ImagePixels.java:17]: I don't see where you instantiate this.

File: src/main/java/yahav/photomosaic/ImagePixels.java (hunk 5 above)
Comment (andrewoid) [reply src/main/java/yahav/photomosaic/ImagePixels.java:33]: Nevermind. I think I was momentarily confused as to what SQUARES means.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**